wardrobe/tradable_kitties/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+    DuplicateTradableKittyFound,
+    /// Kitty price is unaltered is not allowed for kitty price update transactions.
+    KittyPriceUnaltered,
+}
+
+impl From<money::ConstraintCheckerError> for TradeableKittyError {
+    fn from(error: money::ConstraintCheckerError) -> Self {
+        TradeableKittyError::MoneyError(error)
+    }
+}
+
+impl From<kitties::ConstraintCheckerError> for TradeableKittyError {
+    fn from(error: kitties::ConstraintCheckerError) -> Self {
+        TradeableKittyError::KittyError(error)
+    }
+}
+
+/// The main constraint checker for the trdable kitty piece. Allows below :
+/// Listing kitty for sale : multiple kiities are allowed, provided input and output in same order.
+/// Delisting kitty from sale : multiple tradable kiities are allowed, provided input and output in same order.
+/// Update kitty price : multiple tradable kiities are allowed, provided input and output in same order.
+/// Update kitty name : multiple tradable kiities are allowed, provided input and output in same order.
+/// Buy tradable kitty :  multiple tradable kiities are not allowed, Only single kiity operation is allowed.
+#[derive(
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Clone,
+    Encode,
+    Decode,
+    Hash,
+    Debug,
+    TypeInfo,
+)]
+pub enum TradableKittyConstraintChecker<const ID: u8> {
+    /// List kitty for sale, means kitty will converted to tradable kitty once transaction is executed
+    ListKittyForSale,
+    /// Delist kitty from sale, means tradable kitty will converted back to kitty  
+    DelistKittyFromSale,
+    /// Update price of tradable kitty.
+    UpdateKittyPrice,
+    // Update name of kitty
+    UpdateKittyName,
+    /// For buying a new kitty from others
+    Buy,
+}
+
+/// Extracts basic kitty data from a list of dynamically typed TradableKitty data, populating basic kitty data list.
+fn extract_basic_kitty_list(
+    tradable_kitty_data: &[DynamicallyTypedData],
+    kitty_data_list: &mut Vec<DynamicallyTypedData>,
+) -> Result<(), TradeableKittyError> {
+    for utxo in tradable_kitty_data {
+        if let Ok(tradable_kitty) = utxo.extract::<TradableKittyData>() {
+            kitty_data_list.push(tradable_kitty.kitty_basic_data.clone().into());
+        } else {
+            return Err(TradeableKittyError::BadlyTyped);
+        }
+    }
+    Ok(())
+}
+
+/// Checks if buying the kitty is possible of not. It depends on Money piece to validate spending of coins.
+fn check_can_buy<const ID: u8>(
+    input_data: &[DynamicallyTypedData],
+    output_data: &[DynamicallyTypedData],
+) -> Result<(), TradeableKittyError> {
+    let mut input_coin_data: Vec<DynamicallyTypedData> = Vec::new();
+    let mut output_coin_data: Vec<DynamicallyTypedData> = Vec::new();
+    let mut input_ktty_to_be_traded: Option<TradableKittyData> = None;
+    let mut output_ktty_to_be_traded: Option<TradableKittyData> = None;
```

Comment:
Variable names are missing the i in kitty.

Reply:
fixed